sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -539,6 +541,15 @@ def run_pipeline(self, pipeline, options):
     # Get a Dataflow API client and set its options
     self.dataflow_client = apiclient.DataflowApplicationClient(options)
 
+    if self.job.options.view_as(GoogleCloudOptions).upload_graph:
```

Comment:
Similarly, this staging can happen in apiclient.py if the experiment is present.

Reply:
Re hiding it in apiclient - in first commit I've implemented it that way then I've noticed java implementation where it is implemented  in runner instead of apiclient and decided to move.  Will move again and rewrite tests.